apisix/plugins/grpc-transcode/response.lua

```diff
@@ -99,7 +99,10 @@ return function(ctx, proto, service, method, pb_option, show_status_in_body, sta
 
     -- handle error response after the last response chunk
     if ngx.status >= 300 and show_status_in_body then
-        return handle_error_response(status_detail_type)
+        local pb_old_state = pb.state(proto.pb_state)
+        local ret = handle_error_response(status_detail_type)
+        pb.state(pb_old_state)
```

Comment:
Why not do this in `handle_error_response` function, `around pd.decode` like other place.